### PR TITLE
Increase logging information

### PR DESCRIPTION
**Pull request by osama-salman99**
* [x] Code is up-to-date with the `main` branch.
* [x] You've successfully built and run the tests locally.
* [x] There are new or updated unit tests validating the change.

### :pencil: Description
- Increase logging information

### :link: Related Issues
N/A

### CODE REVIEW

File: src/main/kotlin/com/expediagroup/sdk/core/client/openapi/OpenApiStub.kt [hunk 1]
```diff
@@ -35,12 +35,9 @@ import io.ktor.http.toURI
 abstract class OpenApiStub(
     private val clientConfiguration: ClientConfiguration = ClientConfiguration.EMPTY
 ) {
+    private val log = OpenWorldLoggerFactory.getLogger(javaClass)
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/client/openapi/OpenApiStub.kt]: Why do we convert it to not static filed? It's common for loggers to be static.

File: src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt [hunk 2]
```diff
@@ -43,8 +50,12 @@ internal class ConfigurationCollector private constructor(providers: Configurati
...
-    override val key: String by lazy { providers.firstOf { it.key } }
-    override val secret: String by lazy { providers.firstOf { it.secret } }
-    override val endpoint: String by lazy { providers.firstOf { it.endpoint } }
-    override val authEndpoint: String by lazy { providers.firstOf { it.authEndpoint } }
+    val key: String by lazy { providers.firstOf { provider -> provider.key.also { logConfigurationRetrieval(provider, KEY) } } }
+    val secret: String by lazy { providers.firstOf { provider -> provider.secret.also { logConfigurationRetrieval(provider, SECRET) } } }
+    val endpoint: String by lazy { providers.firstOf { provider -> provider.endpoint.also { logConfigurationRetrieval(provider, ENDPOINT) } } }
+    val authEndpoint: String by lazy { providers.firstOf { provider -> provider.authEndpoint.also { logConfigurationRetrieval(provider, AUTH_ENDPOINT) } } }
+
+    private fun logConfigurationRetrieval(provider: ConfigurationProvider, configurationName: String) {
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt]: We may define this as an extension function inside this class (with a short name, e.g., `log`).

File: src/main/kotlin/com/expediagroup/sdk/core/constant/Constant.kt [hunk 3]
```diff
@@ -22,6 +22,20 @@ internal object Constant {
 
     const val BEARER = "Bearer"
 
+    const val KEY = "key"
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/constant/Constant.kt]: 1. Can we have these constants in a new class?
2. We may consider more specific names to `DEFAULT` and `FILE_SYSTEM `.

**Review by mohnoor94 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/expediagroup/sdk/core/client/openapi/OpenApiStub.kt (hunk 1 above)
Comment (osama-salman99) [reply src/main/kotlin/com/expediagroup/sdk/core/client/openapi/OpenApiStub.kt]: For consistency. As Kotlin `object`s cannot have a `companion object`.

File: src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt [hunk 4]
```diff
@@ -16,15 +16,22 @@
...
+import com.expediagroup.sdk.core.constant.provider.LoggingMessageProvider
+import com.expediagroup.sdk.core.plugin.logging.OpenWorldLoggerFactory
 
 /**
  * Configuration collector that collects configuration from all available providers.
  *
  * @param providers A configuration providers queue.
  */
-internal class ConfigurationCollector private constructor(providers: ConfigurationProviderQueue) : ConfigurationProvider {
+internal class ConfigurationCollector private constructor(providers: ConfigurationProviderQueue) {
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt]: Why do you remove the interface? We need this to implement the `ConfigurationProvider` interface. Please undo this change.

File: src/main/kotlin/com/expediagroup/sdk/core/constant/Constant.kt [hunk 5]
```diff
@@ -20,7 +20,7 @@ import java.io.File
 internal object Constant {
     const val EMPTY_STRING = ""
 
-    const val BEARER = "Bearer"
+    const val RUNTIME_CONFIGURATION = "runtime configuration"
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/constant/Constant.kt]: This should be in the new `Configuration` object.

File: src/main/kotlin/com/expediagroup/sdk/core/constant/Configuration.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.expediagroup.sdk.core.constant
+
+internal object Configuration {
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/sdk/core/constant/Configuration.kt]: I would suggest `ConfigurationName` as a file and object name.

**Review by mohnoor94 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt (hunk 4 above)
Comment (osama-salman99) [reply src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt]: This class, as the name suggests, is a collector, not a provider. I also found that it doesn't need to implement the `ConfigurationProvider` interface, it is not used in this context anywhere. Making it implement the `ConfigurationProvider` interface means that we should give it the attribute `name` that is not even used anywhere and it doesn't make sense for this class to have an identifier or name like the other providers. What do you think?

Comment (mohnoor94) [reply src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt]: It is a `ConfigurationProvider` that collects the configurations from other providers. Furthermore, we needed to make it implement the `ConfigurationProvider` interface to make sure it's up-to-date with any changes in the providers so that one should not need to access any `ConfigurationProvider` implementations directly (but via the `ConfigurationCollector`).

For it to have a `name`, we have 2 options:
1. Let it have a `name`, even if it's not used.
2. Revert back to the solution we discussed earlier of having an external map of the names instead of adding a `name` property to the interface.

I prefer the first option.

Comment (osama-salman99) [reply src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt]: Alright, that makes sense. I will be sticking with the first option and give it a name

**Review by mohnoor94 — APPROVED** [review]
